lightning/connection_commando.go

```diff
@@ -0,0 +1,147 @@
+package lightning
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/bolt-observer/agent/lnsocket"
+)
+
+// ClnCommandoConnection struct
+type ClnCommandoConnection struct {
+	ln   *lnsocket.LN
+	addr string
+	rune string
+
+	ClnConnection
+}
+
+// Compile time check for the interface
+var _ ClnConnectionAPI = &ClnCommandoConnection{}
+
+// MakeCommandoConnection create a new CLN connection
+func MakeCommandoConnection(addr string, rune string, timeout time.Duration) *ClnCommandoConnection {
+	ret := &ClnCommandoConnection{}
+
+	ret.addr = addr
+	ret.rune = rune
+	ret.ln = lnsocket.MakeLN(timeout)
+
+	return ret
+}
+
+func convertArgs(param any) string {
```

Comment:
its easier to read if helper methods are below its usage. Also makes sense to se API (public methods) first